### PR TITLE
🗑️ cleanup after adding logstreaming

### LINKED ISSUE #5059: Add tests for rabbitmq in the servicelib.fastapi
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Planned changes

_No response_

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Enable unit tests for rabbitQM part in service lib
- Factor out `is_pdb_enabled` pytest fixture to pytest-simcore
- general cleanup needed after introducing log streaming endpoint

## Related issue/s
- resolves #5059 

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Comment by YuryHrytsuk** [comment]
Don't forget to go through get `Dev Checklist`

File: packages/service-library/src/servicelib/fastapi/rabbitmq.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ async def reconnect(app: FastAPI):
 def setup_rabbit(
     app: FastAPI,
     *,
-    settings: RabbitSettings,
+    settings: RabbitSettings | None,
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/fastapi/rabbitmq.py:45]: Question: do we need it optional? It is kind of strange to setup rabbit without having any settings

File: services/api-server/tests/unit/test_services_rabbitmq.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def app_environment(
     mocker: MockerFixture,
 ) -> EnvVarsDict:
     # do not init other services
+    mocker.stopall()
```
Comment (sanderegg) [inline services/api-server/tests/unit/test_services_rabbitmq.py:59]: this is a bit suspicious. why do we need this?

**Review by sanderegg — APPROVED** [review]
some questions, thanks!

File: packages/service-library/src/servicelib/fastapi/rabbitmq.py (hunk 1 above)
Comment (bisgaard-itis) [reply packages/service-library/src/servicelib/fastapi/rabbitmq.py:45]: Good point, I corrected this.

File: services/api-server/tests/unit/test_services_rabbitmq.py (hunk 2 above)
Comment (bisgaard-itis) [reply services/api-server/tests/unit/test_services_rabbitmq.py:59]: Yes, I didn't see how else to do this. The point is that I only need the rabbitmq client for very few tests, so in the `app_environment` fixture in `conftest.py` I mock the fcn which sets up the rabbitmq client [here](https://github.com/ITISFoundation/osparc-simcore/blob/38e1f63203389f3867a590fa81baf4f229ebc03d/services/api-server/tests/unit/conftest.py#L65). And apparently when I override fixtures, the toplevel fixture is called first. Hence, when I arrive here the rabbitmq client is mocked. Hence, I need to remove that mock again. Do you see a better way to do this?

Comment (bisgaard-itis) [reply services/api-server/tests/unit/test_services_rabbitmq.py]: Sorry for the noise, I fixed this.

**Review by pcrespov — APPROVED** [review]

**Pull request merged**